promql/engine.go

```diff
@@ -1795,17 +1861,31 @@ func (ev *evaluator) vectorSelectorSingle(it *storage.MemoizedSeriesIterator, no
 	return t, v, h, true
 }
 
-var pointPool zeropool.Pool[[]Point]
+var (
+	fPointPool zeropool.Pool[[]FPoint]
+	hPointPool zeropool.Pool[[]HPoint]
+)
 
-func getPointSlice(sz int) []Point {
-	if p := pointPool.Get(); p != nil {
+func getFPointSlice(sz int) []FPoint {
+	if p := fPointPool.Get(); p != nil {
 		return p
 	}
-	return make([]Point, 0, sz)
+	return make([]FPoint, 0, sz)
 }
 
-func putPointSlice(p []Point) {
-	pointPool.Put(p[:0])
+func putFPointSlice(p []FPoint) {
+	fPointPool.Put(p[:0])
```

Comment:
It's definitely more robust to check for `p==nil`. Since this wasn't done before, either, I'll add it in a separate commit. (It's unrelated to the changes in the other commits.)